services/can_logger/can_logger.py

```diff
@@ -32,22 +32,65 @@ def write_to_csv(wr_buff, can_bus):
 
     print(f'{can_bus}: Wrote successfully to CSV')
 
-def rotate_log():
-    global firstRun
-    t = time.strftime('%H',time.localtime())
-    # Check if enough time has passed to rotate the log
-    if(firstRun):
-        timestamp = time.strftime('%m-%d-%Y-%H-%M',time.localtime())
-        last_time = t
-        firstRun = False
-    elif(t == last_time):
-        pass
+def rotate_log(can_bus, interval, frequency):
+    last_time = 0
+    f = 0
+    firstRun = True
+    # Check for a valid interval. If valid, set dt (update interval) to its
+    # equivalent in seconds, i.e., 60 seconds for minute rotation and 3600s for
+    # hourly rotation.
+    if(interval == 'S'):
+        dt = 1
+        interval_str = "second(s)"
+    if(interval == 'M'):
+        dt = 60
+        interval_str = 'minute(s)'
+    elif(interval == 'H'):
+        dt = 3600
+        interval_str = 'hour(s)'
+    elif(interval == 'D'):
+        dt = 86400
+        interval_str = 'day(s)'
     else:
-        last_time = t
-        # Save new timestamp
-        timestamp = time.strftime('%m-%d-%Y-%H-%M',time.localtime())
+        # Set interval to blank if the specified interval is not valid
+        interval = ''
+    # Enter to the loop if the interval is not blank
+    while(interval):
```

Comment:
Assuming that this is equivalent to `while interval != '':`, I would prefer the explicit comparison for clarity. As it is, it implies to me that interval is a boolean